core/__init__.py

```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python3.6
 # -*- coding: utf-8 -*-
 
+"""Core modules that are used exclusively by :py:mod:`nbuild` or the compilation library (:py:mod:`stdlib`).
+
+The content of the core module shouldn't be used by a build manifest.
```

Comment:
I think "should not" would be more appropriate here (this is also valid for other occurrences of "shouldn't" in this PR).